lib/matplotlib/cbook/__init__.py

```diff
@@ -199,48 +199,61 @@ def __setstate__(self, state):
 
     def connect(self, s, func):
         """Register *func* to be called when signal *s* is generated."""
-        self._func_cid_map.setdefault(s, {})
+        signal = s
+        self._func_cid_map.setdefault(signal, {})
         proxy = _weak_or_strong_ref(func, self._remove_proxy)
-        if proxy in self._func_cid_map[s]:
-            return self._func_cid_map[s][proxy]
+        if proxy in self._func_cid_map[signal]:
+            return self._func_cid_map[signal][proxy]
+
         cid = next(self._cid_gen)
-        self._func_cid_map[s][proxy] = cid
-        self.callbacks.setdefault(s, {})
-        self.callbacks[s][cid] = proxy
+        self._func_cid_map[signal][proxy] = cid
+        self.callbacks.setdefault(signal, {})
+        self.callbacks[signal][cid] = proxy
         return cid
 
     # Keep a reference to sys.is_finalizing, as sys may have been cleared out
     # at that point.
-    def _remove_proxy(self, proxy, *, _is_finalizing=sys.is_finalizing):
+    def _remove_proxy(self, proxy, _is_finalizing=sys.is_finalizing):
```

Comment:
why did you remove the keyword-onlyness?

Reply:
Because it's a private function, and it is unused. THe callback of weakref is only one argument, no need extra 

Same for `_is_finalizing=sys.is_finalizing`, i really think it is pointless.

All this stuffs was probably useful 10 years ago? But to me, now it's noise.

Reply:
The point is to avoid other internal places from ever accidentally passing the argument.
`_is_finalizing=sys.is_finalizing` is not pointless, this may be called during interpreter shutdown etc. as documented in the comment just above; this was added in response to an actual bug which you can search in the tracker.

Reply:
Calling private function is a bug on it's own.

And i saw the comment, about finalized. But the only meaning of this function is to be called during the finalization. If there is a bug here, it's a bug inside python interpreter itself. It's probably better to add a link to this python issue at this place of the code.

Reply:
> Calling private function is a bug on it's own.

Not if *we* call it internally.

> it's a bug inside python interpreter itself.

No, this is documented behavior (https://docs.python.org/3/reference/datamodel.html#object.__del__, https://docs.python.org/3/glossary.html#term-interpreter-shutdown).

Please be careful before making overly broad statements.

Reply:
Well, `_remove_proxy` most not be called outside of this class by anything. You don't want it to be triggered twice anyway.

For finalizing, i am aware of this things. I just tell that the only purpose of this function is to be called during the finalizing stage. And there is no warn about a special way to use this function in the python documentation, such warning you would set with blinking style, cause it's the only use case of this function. So, i still don't think it is needed.

But i understand you tend to be conservative.

Reply:
You haven't begun to interact with any of the conservative members of the dev team!

However, please try and defer to @anntzer's suggestions, even if they are not your own stylistic choice.  He is the Principal Engineer on the project, and was chosen for that role because of his deep knowledge of both python and matplotlib and the respect the rest of the dev teams has for that expertise.  Its possible this could be merged despite his concerns, but it will go a lot faster if you address them :wink:.

Reply:
Don't worry, i just argue because it's an interesting question on it's own.

I use to work with many commits in such case, cause most of them are not directly connected to the problem, and i know it. There is no problem for me to rework that stuff, but i have to find the time.

Reply:
I agree with @anntzer that we should keep this as a keyword-only.

Reply:
I have removed my commit relative to the remove of `*`